### PR TITLE
Autoname PE import thunks

### PR DESCRIPTION
**Pull request by pelijah**
Related: #11759

### CODE REVIEW

**Comment by radare** [comment]
I would prefer to have different prs for this

File: libr/core/canal.c [hunk 1]
```diff
@@ -437,6 +437,52 @@ R_API void r_core_anal_autoname_all_fcns(RCore *core) {
...
 
+static bool autoname_pe_imp_thunk(RFlagItem *flag, void *u) {
+	RFlagItem *fcn_flag;
+	RCore *core = (RCore *) u;
+	RList *xrefs;
+	RAnalRef *ref;
+	RAnalFunction *fcn;
+	char *name;
+
+	xrefs = r_anal_xrefs_get (core->anal, flag->offset);
```
Comment (radare) [inline libr/core/canal.c]: define and assign in the same line

File: libr/core/canal.c [hunk 2]
```diff
@@ -437,6 +437,52 @@ R_API void r_core_anal_autoname_all_fcns(RCore *core) {
...
+	if (!xrefs || r_list_length (xrefs) > 1) {
+		goto beach;
+	}
+	ref = r_list_first (xrefs);
```
Comment (radare) [inline libr/core/canal.c]: same here

File: libr/core/canal.c [hunk 3]
```diff
@@ -437,6 +437,52 @@ R_API void r_core_anal_autoname_all_fcns(RCore *core) {
...
+beach:
+	r_list_free (xrefs);
+	return true;
+}
+
+R_API void r_core_anal_autoname_all_pe_imp_thunk(RCore *core) {
+	RFlagItem *imp_flag;
+	RSpace *imp_space;
+
+	imp_space = r_flag_space_get (core->flags, "imports");
```
Comment (radare) [inline libr/core/canal.c]: define and assign in the same line

**Comment by ITAYC0HEN** [comment]
Very nice! We soon will be able to say goodbye to aan!
Can you please add comments describing the logic and algorithm?

**Comment by radare** [comment]
Uh?

**Comment by ITAYC0HEN** [comment]
Ignores r2 naming convention:

image

**Comment by ITAYC0HEN** [comment]
see: https://github.com/radare/radare2/pull/13124#issuecomment-464352243

**Comment by ITAYC0HEN** [comment]
I think this can be closed since #13124 merged

**Pull request closed without merging**